### ISSUE TITLE
Filenames with UTF-8 characters in file root aren't supported by logger

### ISSUE THREAD

**Issue by dereckson** [saltstack/salt#44406]
### Description of Issue/Question
UTF-8 queer cornercase when a file with UTF-8 characters exists in the file root directory:

`log.trace('roots: Processing %s', abs_path)` doesn't expect abs_path to be an UTF-8 string.

That raises the following exception:
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 53: ordinal not in range(128)

```
  File "/usr/local/lib/python2.7/site-packages/salt/fileserver/__init__.py", line 896, in send
    return getattr(self.fs, cmd)(load)
  File "/usr/local/lib/python2.7/site-packages/salt/fileserver/__init__.py", line 756, in file_list
    ret.update(self.servers[fstr](load))
  File "/usr/local/lib/python2.7/site-packages/salt/fileserver/roots.py", line 421, in file_list
    return _file_lists(load, 'files')
  File "/usr/local/lib/python2.7/site-packages/salt/fileserver/roots.py", line 397, in _file_lists
    _add_to(ret['files'], path, root, files)
  File "/usr/local/lib/python2.7/site-packages/salt/fileserver/roots.py", line 343, in _add_to
    log.trace('roots: Processing %s', abs_path)
  File "/usr/local/lib/python2.7/site-packages/salt/log/mixins.py", line 35, in trace
    self.log(getattr(logging, 'TRACE', 5), msg, *args, **kwargs)
  File "/usr/local/lib/python2.7/logging/__init__.py", line 1231, in log
    self._log(level, msg, args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/salt/log/setup.py", line 327, in _log
    self, level, msg, args, exc_info=exc_info, extra=extra
  File "/usr/local/lib/python2.7/logging/__init__.py", line 1285, in _log
    record = self.makeRecord(self.name, level, fn, lno, msg, args, exc_info, func, extra)
  File "/usr/local/lib/python2.7/site-packages/salt/log/setup.py", line 361, in makeRecord
    exc_info, func)
  File "/usr/local/lib/python2.7/site-packages/salt/log/setup.py", line 208, in __init__
    self.colormsg = '%s%s%s' % (cmsg, self.getMessage(), reset)
  File "/usr/local/lib/python2.7/logging/__init__.py", line 329, in getMessage
    msg = msg % self.args
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 53: ordinal not in range(128)
```

### Setup
Salt running on Python 2 + a file root directory managed as a Git repository

### Steps to Reproduce Issue

Original issue found with this kind of setup:

```
$ cd /srv/salt
$ git init .
$ git checkout -b <something with an UTF-8 character>
$ salt-call --local state.apply <some state>
```

But `touch /srv/salt/æ` would be enough.

### Versions Report
Salt Version:
           Salt: 2017.7.1
         Jinja2: 2.9.5
         Python: 2.7.14 (default, Oct  3 2017, 01:18:38)
         PyYAML: 3.11

(locale: LANG=en_US.UTF-8)

**Comment by damon-atkins** [comment]
See https://github.com/saltstack/salt/issues/42278 `salt logging with plain str (ascii) with unicode chars causes UnicodeDecodeError PY2`

**Comment by rossengeorgiev** [comment]
What's the output of `locale`? If it doesn't have UTF-8, just do `export LANG=C.UTF-8` and run `salt-call` again.

**Comment by dereckson** [comment]
Locale is set as this:
```
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_ALL=
```

**Comment by damon-atkins** [comment]
Try the fix See #42278 (comment)

**Comment by dereckson** [comment]
A fix is still to test per @damon-atkins comment. There is also a need to tetermine if it's a dupe of the #42278 (still opened) issue.
